### ISSUE TITLE
[Bug]: View Texture crashes game 

### ISSUE THREAD

**Issue by gottlly** [yukieiji/UnityExplorer#18]
### Are you on the latest version of UnityExplorer?

- [X] Yes, I'm on the latest version of UnityExplorer.

### Which release are you using?

BepInEx Unity IL2CPP CoreCLR

### Which game did this occur on?

V Rising

### Describe the issue.

When I select "View Texture" on Texture2ds, the whole thing crashes 

### Relevant log output

```shell
UnityExplorer 4.12.1 initializing...
[UniverseLib] UniverseLib 1.5.6 initializing...
[UniverseLib] Finished UniverseLib initial setup.
Initialized UnityCrashPrevention for: Canvas.get_renderingDisplaySize
[UniverseLib] Can't cache type named MethodInfoStoreGeneric_TryGetAsset_Public_Virtual_Final_New_Boolean_EntityManager_AssetGuid_byref_T_0`1 Error: System.TypeLoadException: Could not load type 'ProjectM.HybridEquipmentStreamingHandler+Singleton' from assembly 'ProjectM, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.RuntimeTypeHandle.GetDeclaringType(RuntimeType type)
   at System.RuntimeType.RuntimeTypeCache.GetEnclosingType()
   at System.RuntimeType.RuntimeTypeCache.GetNameSpace()
   at System.RuntimeType.get_Namespace()
   at UniverseLib.ReflectionUtility.CacheTypes(Assembly asm) in F:\Documents\VisualStudioProject\AmongUs\UnityExplorer\UniverseLib\src\Reflection\ReflectionUtility.cs:line 121
[UniverseLib] Can't cache type named MethodInfoStoreGeneric_TryGetAsset_Public_Virtual_Final_New_Boolean_EntityManager_AssetGuid_byref_T_0`1 Error: System.TypeLoadException: Could not load type 'ProjectM.HybridModelStreamingHandler+Singleton' from assembly 'ProjectM, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.RuntimeTypeHandle.GetDeclaringType(RuntimeType type)
   at System.RuntimeType.RuntimeTypeCache.GetEnclosingType()
   at System.RuntimeType.RuntimeTypeCache.GetNameSpace()
   at System.RuntimeType.get_Namespace()
   at UniverseLib.ReflectionUtility.CacheTypes(Assembly asm) in F:\Documents\VisualStudioProject\AmongUs\UnityExplorer\UniverseLib\src\Reflection\ReflectionUtility.cs:line 121
[UniverseLib] Can't cache type named TypeHandle Error: System.TypeLoadException: Could not load type 'ProjectM.HybridModelStreamingHandler+Singleton' from assembly 'ProjectM, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.RuntimeTypeHandle.GetDeclaringType(RuntimeType type)
   at System.RuntimeType.RuntimeTypeCache.GetEnclosingType()
   at System.RuntimeType.RuntimeTypeCache.GetNameSpace()
   at System.RuntimeType.get_Namespace()
   at UniverseLib.ReflectionUtility.CacheTypes(Assembly asm) in F:\Documents\VisualStudioProject\AmongUs\UnityExplorer\UniverseLib\src\Reflection\ReflectionUtility.cs:line 121
[UniverseLib] Can't cache type named MethodInfoStoreGeneric_TryGetAsset_Public_Virtual_Final_New_Boolean_EntityManager_AssetGuid_byref_T_0`1 Error: System.TypeLoadException: Could not load type 'ProjectM.UIAssetSubSceneLoader_ClientWorld+Singleton' from assembly 'ProjectM, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.RuntimeTypeHandle.GetDeclaringType(RuntimeType type)
   at System.RuntimeType.RuntimeTypeCache.GetEnclosingType()
   at System.RuntimeType.RuntimeTypeCache.GetNameSpace()
   at System.RuntimeType.get_Namespace()
   at UniverseLib.ReflectionUtility.CacheTypes(Assembly asm) in F:\Documents\VisualStudioProject\AmongUs\UnityExplorer\UniverseLib\src\Reflection\ReflectionUtility.cs:line 121
[UniverseLib] Can't cache type named MethodInfoStoreGeneric_TryGetAsset_Public_Virtual_Final_New_Boolean_EntityManager_AssetGuid_byref_T_0`1 Error: System.TypeLoadException: Could not load type 'ProjectM.UIAssetSubSceneLoader_DefaultWorld+Singleton' from assembly 'ProjectM, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.RuntimeTypeHandle.GetDeclaringType(RuntimeType type)
   at System.RuntimeType.RuntimeTypeCache.GetEnclosingType()
   at System.RuntimeType.RuntimeTypeCache.GetNameSpace()
   at System.RuntimeType.get_Namespace()
   at UniverseLib.ReflectionUtility.CacheTypes(Assembly asm) in F:\Documents\VisualStudioProject\AmongUs\UnityExplorer\UniverseLib\src\Reflection\ReflectionUtility.cs:line 121
[UniverseLib] Can't cache type named MulticastDelegateNPublicSealedVoObInUnique Error: System.TypeLoadException: Could not load type 'ProjectM.HybridEquipmentStreamingHandler+Singleton' from assembly 'ProjectM, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.RuntimeTypeHandle.GetDeclaringType(RuntimeType type)
   at System.RuntimeType.RuntimeTypeCache.GetEnclosingType()
   at System.RuntimeType.RuntimeTypeCache.GetNameSpace()
   at System.RuntimeType.get_Namespace()
   at UniverseLib.ReflectionUtility.CacheTypes(Assembly asm) in F:\Documents\VisualStudioProject\AmongUs\UnityExplorer\UniverseLib\src\Reflection\ReflectionUtility.cs:line 121
[UniverseLib] Can't cache type named ObjectNInternalAbstractSealedInPoDeInUnique Error: System.TypeLoadException: Could not load type 'ProjectM.HybridEquipmentStreamingHandler+Singleton' from assembly 'ProjectM, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.RuntimeTypeHandle.GetDeclaringType(RuntimeType type)
   at System.RuntimeType.RuntimeTypeCache.GetEnclosingType()
   at System.RuntimeType.RuntimeTypeCache.GetNameSpace()
   at System.RuntimeType.get_Namespace()
   at UniverseLib.ReflectionUtility.CacheTypes(Assembly asm) in F:\Documents\VisualStudioProject\AmongUs\UnityExplorer\UniverseLib\src\Reflection\ReflectionUtility.cs:line 121
[UniverseLib] Can't cache type named MulticastDelegateNPublicSealedVoObInUnique Error: System.TypeLoadException: Could not load type 'ProjectM.HybridEquipmentStreamingHandler+Singleton' from assembly 'ProjectM, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.RuntimeTypeHandle.GetDeclaringType(RuntimeType type)
   at System.RuntimeType.RuntimeTypeCache.GetEnclosingType()
   at System.RuntimeType.RuntimeTypeCache.GetNameSpace()
   at System.RuntimeType.get_Namespace()
   at UniverseLib.ReflectionUtility.CacheTypes(Assembly asm) in F:\Documents\VisualStudioProject\AmongUs\UnityExplorer\UniverseLib\src\Reflection\ReflectionUtility.cs:line 121
[UniverseLib] Can't cache type named ObjectNInternalAbstractSealedInPoDeInUnique Error: System.TypeLoadException: Could not load type 'ProjectM.HybridEquipmentStreamingHandler+Singleton' from assembly 'ProjectM, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.RuntimeTypeHandle.GetDeclaringType(RuntimeType type)
   at System.RuntimeType.RuntimeTypeCache.GetEnclosingType()
   at System.RuntimeType.RuntimeTypeCache.GetNameSpace()
   at System.RuntimeType.get_Namespace()
   at UniverseLib.ReflectionUtility.CacheTypes(Assembly asm) in F:\Documents\VisualStudioProject\AmongUs\UnityExplorer\UniverseLib\src\Reflection\ReflectionUtility.cs:line 121
[UniverseLib] Loaded Unhollowed modules in 3.4190001 seconds.
[UniverseLib] Setup deobfuscation cache in 0.09900001 seconds.
[UniverseLib] Initialized Legacy Input support
[UniverseLib] 	 IL2CPP method has no corresponding pointer, aborting patch of UnityEngine.AssetBundle.UnloadAllAssetBundles
[UniverseLib] Loaded modern bundle for Unity 2022.3.23f1
[UniverseLib] This game does not ship with the 'UI/Default' shader, using manual Default Shader...
[UniverseLib] UniverseLib 1.5.6 initialized.
Creating UI...

UnityExplorer 4.12.1 (IL2CPP) initialized.
```

**Comment by yukieiji** [comment]
This game startup log isn't a bug(I don't even think it has anything to do with this issues)
Can you attach all logs?

**Comment by gottlly** [comment]
it was in D:\SteamLibrary\steamapps\common\VRising\BepInEx\plugins\sinai-dev-UnityExplorer\Logs so I grabbed the latest file. I didnt make that folder myself, so I assumed it was what UE had to give me, even if not very informative... I have another 9 logs that look about the same i think 

![image](https://github.com/yukieiji/UnityExplorer/assets/171308556/9636bdcd-acb1-4048-9ed8-3d7228cba4ed)

ill individually attach them here too 
[UnityExplorer 2024-06-01 194810Z.txt](https://github.com/user-attachments/files/15524092/UnityExplorer.2024-06-01.194810Z.txt)
[UnityExplorer 2024-06-01 210812Z.txt](https://github.com/user-attachments/files/15524093/UnityExplorer.2024-06-01.210812Z.txt)
[UnityExplorer 2024-06-01 211456Z.txt](https://github.com/user-attachments/files/15524094/UnityExplorer.2024-06-01.211456Z.txt)
[UnityExplorer 2024-06-01 211931Z.txt](https://github.com/user-attachments/files/15524095/UnityExplorer.2024-06-01.211931Z.txt)
[UnityExplorer 2024-06-01 212119Z.txt](https://github.com/user-attachments/files/15524096/UnityExplorer.2024-06-01.212119Z.txt)
[UnityExplorer 2024-06-01 213115Z.txt](https://github.com/user-attachments/files/15524097/UnityExplorer.2024-06-01.213115Z.txt)
[UnityExplorer 2024-06-01 220758Z.txt](https://github.com/user-attachments/files/15524098/UnityExplorer.2024-06-01.220758Z.txt)
[UnityExplorer 2024-06-01 220822Z.txt](https://github.com/user-attachments/files/15524099/UnityExplorer.2024-06-01.220822Z.txt)
[UnityExplorer 2024-06-01 221159Z.txt](https://github.com/user-attachments/files/15524100/UnityExplorer.2024-06-01.221159Z.txt)
[UnityExplorer 2024-06-01 221337Z.txt](https://github.com/user-attachments/files/15524101/UnityExplorer.2024-06-01.221337Z.txt)

I dont know much about... much, really. So admittedly it might be user error on my part

**Comment by yukieiji** [comment]
Please attach a Log of the BepInEx.

**Comment by gottlly** [comment]
These are the two log files I have inside the BepInEx folder from my attempt just now. 
[ErrorLog.log](https://github.com/user-attachments/files/15524276/ErrorLog.log)
[LogOutput.log](https://github.com/user-attachments/files/15524277/LogOutput.log)

**Comment by yukieiji** [comment]
I am seeing an error output due to accessing a protected area, so I think this is the crash caused by that.
Does it crash when clicking on any texture?

**Comment by gottlly** [comment]
The one i clicked was a texture for a dlc skin, I realized and decided to check for the default skin, which also crashed. 
[ErrorLog.log](https://github.com/user-attachments/files/15524386/ErrorLog.log)

I then tried to view an UI element, as other mods have said they were able to get/modify those depending on the case. crashed again 
[ErrorLog.log](https://github.com/user-attachments/files/15524391/ErrorLog.log)

**Comment by yukieiji** [comment]
Are there any anti-cheats, etc. activated? I feel like it's protected and can't be referenced by UE

**Comment by gottlly** [comment]
Nothing I've set up myself, though this game does have pvp so it wouldn't surprise me. Are these anti-cheats unavoidable when modding? A lot of this game is just building and aesthetic things and I was excited to make content for it. Went into a bit of a hyperfixation with it the last week and a half. 

I managed to download this same texture with UABEA, but couldn't alter it properly (just swapping the file w/ it resulted in a completely transparent model). AssetRipper is able to navigate it / export it too- I understand if tapping into a protected file like this just isnt what UE is for, just thought id open the thread and try
